source/simulator/stokes_matrix_free.cc

```diff
@@ -1176,16 +1176,26 @@ namespace aspect
     const unsigned int block_p = (sim.parameters.include_melt_transport) ?
                                  sim.introspection.variable("fluid pressure").block_index
                                  : sim.introspection.block_indices.pressure;
+
+    LinearAlgebra::BlockVector distributed_stokes_solution (sim.introspection.index_sets.stokes_partitioning,
+                                                            sim.mpi_communicator);
+    // extract Stokes parts of rhs vector
+    LinearAlgebra::BlockVector distributed_stokes_rhs(sim.introspection.index_sets.stokes_partitioning,
+                                                      sim.mpi_communicator);
+
+    distributed_stokes_rhs.block(block_vel) = sim.system_rhs.block(block_vel);
+    distributed_stokes_rhs.block(block_p) = sim.system_rhs.block(block_p);
```

Comment:
why do you need to copy these vectors? The `sim.system_rhs` is already a completely distributed vector

Reply:
It is a Trilinos BlockVector, and we need a `dealii::distributed::BlockVector`. Also, `sim.system_rhs` contains more blocks than just the stokes blocks.